DuckDuckGo/AutomationServer.swift

```diff
@@ -0,0 +1,340 @@
+        }
+
+        // Get url parameter from path
+        // GET / HTTP/1.1
+        if #available(iOS 16.0, *) {
+            let path = /^(GET|POST) (\/[^ ]*) HTTP/
+            if let match = stringContent.firstMatch(of: path) {
+                Logger.automationServer.info("Path: \(match.2)")
+                // Convert the path into a URL object
+                guard let url = URLComponents(string: String(match.2)) else {
+                    Logger.automationServer.error("Invalid URL: \(match.2)")
+                    return // Or handle the error appropriately
+                }
+                switch url.path {
+                case "/navigate":
+                    self.navigate(on: connection, url: url)
+                case "/execute":
+                    self.execute(on: connection, url: url)
+                case "/getUrl":
+                    self.respond(on: connection, response: self.main.currentUrl() ?? "")
+                case "/getWindowHandles":
+                    self.getWindowHandles(on: connection, url: url)
+                case "/closeWindow":
+                    self.closeWindow(on: connection, url: url)
+                case "/switchToWindow":
+                    self.switchToWindow(on: connection, url: url)
+                case "/newWindow":
+                    self.newWindow(on: connection, url: url)
+                case "/getWindowHandle":
+                    self.getWindowHandle(on: connection, url: url)
+                default:
+                    self.respondError(on: connection, error: "unknown")
+                }
+            } else {
+                self.respondError(on: connection, error: "unknown method")
+            }
+        } else {
+            self.respondError(on: connection, error: "unhandled")
+        }
+    }
+
+    @MainActor
+    func navigate(on connection: NWConnection, url: URLComponents) {
+        let navigateUrlString = getQueryStringParameter(url: url, param: "url") ?? ""
+        let navigateUrl = URL(string: navigateUrlString)!
+        self.main.loadUrl(navigateUrl)
+        self.respond(on: connection, response: "done")
+    }
+
+    @MainActor
+    func execute(on connection: NWConnection, url: URLComponents) {
+        let script = getQueryStringParameter(url: url, param: "script") ?? ""
+        var args: [String: String] = [:]
+        // json decode args
+        if let argsString = getQueryStringParameter(url: url, param: "args") {
+            if let argsData = argsString.data(using: .utf8) {
+                do {
+                    let jsonDecoder = JSONDecoder()
+                    args = try jsonDecoder.decode([String: String].self, from: argsData)
+                } catch {
+                    self.respondError(on: connection, error: error.localizedDescription)
+                }
+            } else {
+                self.respondError(on: connection, error: "Unable to decode args")
+            }
+        }
+        Task {
```

Comment:
This task will still run even if you've responded error above - is that intentional?  If not, change the above code to use guards and return out.